hledger-lib/Hledger/Reports/ReportTypes.hs

```diff
@@ -157,10 +152,11 @@ data CompoundPeriodicReport a b = CompoundPeriodicReport
 
 -- | Description of one subreport within a compound balance report.
 -- Part of a "CompoundBalanceCommandSpec", but also used in hledger-lib.
-data CBCSubreportSpec = CBCSubreportSpec
+data CBCSubreportSpec a = CBCSubreportSpec
   { cbcsubreporttitle          :: String
   , cbcsubreportquery          :: Journal -> Query
-  , cbcsubreportnormalsign     :: NormalSign
+  , cbcsubreportoptions        :: ReportOpts -> ReportOpts
+  , cbcsubreporttransform      :: PeriodicReport DisplayName MixedAmount -> PeriodicReport a MixedAmount
```

Comment:
Would you mind adding brief explanatory haddocks for the fields here, at least the new cbcsubreportoptions and cbcsubreporttransform.

Reply:
Done.